### PR TITLE
test: some branches in freshchat service

### PR DESCRIPTION
**Pull request by suyashpatil78**
[workerB](https://workerb.linearb.io/v2/badge/collaboration-page?magicLinkId=smiwBz2)
### <samp>🤖 Generated by Copilot at 3c75351</samp>

This pull request adds more unit tests for the `FreshChatService` class and refactors its code to use a new `getWindow` method. This makes the code more readable, maintainable, and testable.

### <samp>🤖 Generated by Copilot at 3c75351</samp>

> _We face the doom of the widget settings_
> _We test the code with fakeAsync and tick_
> _We abstract the window with a method_
> _We call the getWindow to avoid the dread_

### <samp>🤖 Generated by Copilot at 3c75351</samp>

*  Refactor `FreshChatService` methods to use `getWindow` method for accessing window object ([link](https://github.com/fylein/fyle-mobile-app/pull/2509/files?diff=unified&w=0#diff-5cd10d8760f5baff09c3d8e6d24919c3070e28033e2eb0afdf12995afd1c6300L46-R55), [link](https://github.com/fylein/fyle-mobile-app/pull/2509/files?diff=unified&w=0#diff-5cd10d8760f5baff09c3d8e6d24919c3070e28033e2eb0afdf12995afd1c6300L72-R76), [link](https://github.com/fylein/fyle-mobile-app/pull/2509/files?diff=unified&w=0#diff-5cd10d8760f5baff09c3d8e6d24919c3070e28033e2eb0afdf12995afd1c6300L82-R96), [link](https://github.com/fylein/fyle-mobile-app/pull/2509/files?diff=unified&w=0#diff-5cd10d8760f5baff09c3d8e6d24919c3070e28033e2eb0afdf12995afd1c6300L98-R102), [link](https://github.com/fylein/fyle-mobile-app/pull/2509/files?diff=unified&w=0#diff-5cd10d8760f5baff09c3d8e6d24919c3070e28033e2eb0afdf12995afd1c6300L104-R108), [link](https://github.com/fylein/fyle-mobile-app/pull/2509/files?diff=unified&w=0#diff-5cd10d8760f5baff09c3d8e6d24919c3070e28033e2eb0afdf12995afd1c6300L112-R116))
* Add `getWindow` method to `FreshChatService` class to encapsulate window object access and make it testable ([link](https://github.com/fylein/fyle-mobile-app/pull/2509/files?diff=unified&w=0#diff-5cd10d8760f5baff09c3d8e6d24919c3070e28033e2eb0afdf12995afd1c6300R30-R33))
* Add new test cases for `FreshChatService` in `fresh-chat.service.spec.ts` to cover different scenarios of initiating live chat support and destroying the widget ([link](https://github.com/fylein/fyle-mobile-app/pull/2509/files?diff=unified&w=0#diff-752abd599db5b5f6925f6321a5033bcf7b09653adb3db8f2fda6560f9a27fdccR77-R183))
* Import `fakeAsync` and `tick` functions from `@angular/core/testing` to enable testing of asynchronous code in `fresh-chat.service.spec.ts` ([link](https://github.com/fylein/fyle-mobile-app/pull/2509/files?diff=unified&w=0#diff-752abd599db5b5f6925f6321a5033bcf7b09653adb3db8f2fda6560f9a27fdccL1-R1))
* Import `orgUserSettingsData2` mock data and `FreshChatWidget` model to use in new test cases in `fresh-chat.service.spec.ts` ([link](https://github.com/fylein/fyle-mobile-app/pull/2509/files?diff=unified&w=0#diff-752abd599db5b5f6925f6321a5033bcf7b09653adb3db8f2fda6560f9a27fdccL8-R10))

## Clickup
app.clickup.com

## Code Coverage
Please add code coverage here

## UI Preview
Please add screenshots for UI change

### CODE REVIEW

File: src/app/core/services/fresh-chat.service.ts [hunk 1]
```diff
@@ -27,6 +27,10 @@ export class FreshChatService {
     private networkService: NetworkService
   ) {}
 
+  getWindow(): Window {
```
Comment (OmkarJ13) [inline src/app/core/services/fresh-chat.service.ts:30]: Why refactor window into a separate method?

**Review by OmkarJ13 — APPROVED** [review]

File: src/app/core/services/fresh-chat.service.ts (hunk 1 above)
Comment (suyashpatil78) [reply src/app/core/services/fresh-chat.service.ts:30]: To make it testable, this was necessary step. As then we can mock return values

Comment (OmkarJ13) [reply src/app/core/services/fresh-chat.service.ts:30]: Ah ok, would injecting the `window` object help here? That approach would be cleaner ig.
Just do, `private window: Window` in the constructor of the service, and inject a spy in the test code.

Comment (suyashpatil78) [reply src/app/core/services/fresh-chat.service.ts:30]: If we inject window object as spy we will not be able to change its method.
For eg. we have test for `window.fcWidget.open` and also test if `window.fcWidget = undefined`
which is not possible if we inject it as service. Are you pointing out something else or have any walkaround for this

**Review by Julias0 — APPROVED** [review]

**Pull request merged**